From d9e9261892d47fc806a5465d96bfd99ef97b70d8 Mon Sep 17 00:00:00 2001
From: Martin Fick <mfick@codeaurora.org>
Date: Fri, 7 Dec 2012 16:23:47 -0700
Subject: [PATCH] Fix display updating for dashboards

When I refactored the DashboardTable out of the CustomDashboardScreen
the Screen's display got called from the onLoad() method.  Move it
to the table's onFinishDisplay() method which should get called after
the RPC completes instead of directly after firing the RPC.

Change-Id: I33e629f1986c9999f3420be0a79608f1f60229e6
---
 .../client/changes/CustomDashboardScreen.java      |    5 +++++
 .../client/changes/ProjectDashboardScreen.java     |    5 +++++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CustomDashboardScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CustomDashboardScreen.java
index 1c3ecf7..840ebaa 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CustomDashboardScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/CustomDashboardScreen.java
@@ -30,6 +30,11 @@ public class CustomDashboardScreen extends Screen implements ChangeListScreen {
       @Override
       protected void onLoad() {
         super.onLoad();
+      }
+
+      @Override
+      public void finishDisplay() {
+        super.finishDisplay();
         display();
       }
     };
diff --git a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ProjectDashboardScreen.java b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ProjectDashboardScreen.java
index 76d395b..899e86b 100644
--- a/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ProjectDashboardScreen.java
+++ b/gerrit-gwtui/src/main/java/com/google/gerrit/client/changes/ProjectDashboardScreen.java
@@ -36,6 +36,11 @@ public class ProjectDashboardScreen extends ProjectScreen implements ChangeListS
       @Override
       protected void onLoad() {
         super.onLoad();
+      }
+
+      @Override
+      public void finishDisplay() {
+        super.finishDisplay();
         display();
       }
     };
-- 
1.7.8

